### PR TITLE
Add tinymce editors to article title fields

### PR DESCRIPTION
**Pull request by ajnyga**
See https://github.com/pkp/pkp-lib/pull/3265 for old pr.

work in progress...

### CODE REVIEW

File: templates/form/textarea.tpl [hunk 1]
```diff
@@ -21,18 +26,19 @@
...
 	<span id="{$FBV_name|escape}-localization-popover-container{$uniqId}" class="localization_popover_container">
 		{strip}
-			<textarea id="{$FBV_id|escape}-{$formLocale|escape}{$uniqId}" {$FBV_textAreaParams}
+			<{$textareaElement} id="{$FBV_id|escape}-{$formLocale|escape}{$uniqId}" {$FBV_textAreaParams}
 				rows="{$FBV_rows|escape}"
 				cols="{$FBV_cols|escape}"
-				class="localizable {$FBV_class} {$FBV_height}{if $FBV_validation} {$FBV_validation|escape}{/if}{if $formLocale != $currentLocale} locale_{$formLocale|escape}{/if}{if $FBV_rich && !$FBV_disabled} richContent{if $FBV_rich==="extended"} extendedRichContent{/if}{/if}"
+				class="localizable {$FBV_class} {$FBV_height}{if $FBV_validation} {$FBV_validation|escape}{/if}{if $formLocale != $currentLocale} locale_{$formLocale|escape}{/if}{if $FBV_rich && !$FBV_disabled} richContent{if $FBV_rich==="extended"} extendedRichContent{/if}{if $FBV_rich==="oneline"} onelineRichContent{/if}{/if}"
+ 				{if $FBV_disabled} disabled="disabled"{/if}
 				{if $FBV_disabled} disabled="disabled"{/if}
```
Comment (NateWr) [inline templates/form/textarea.tpl]: Looks like there are two identical lines here now.

Comment (ajnyga) [reply templates/form/textarea.tpl]: must be something that happened when I committed the code, I will send a fix for this later today...

File: js/controllers/form/MultilingualInputHandler.js [hunk 2]
```diff
@@ -247,6 +246,7 @@
 			// html source, for example (both actions open a new window).
 			if ($(tinyMCEObject.getContainer()).find('iframe').attr('id') ==
 					$(document.activeElement).attr('id')) {
+				$(document.activeElement).find('div.mce-tinymce-inline').hide();
```
Comment (NateWr) [inline js/controllers/form/MultilingualInputHandler.js]: What's going on in this code? I can see you're hiding the inline editor, but as far as I can tell, that's when the field is still the active element. I must be misunderstanding something here so just looking for some clarity.

Comment (ajnyga) [reply js/controllers/form/MultilingualInputHandler.js]: See my comments here: https://github.com/pkp/pkp-lib/issues/2564#issuecomment-377704207
In short that is an attempt to solve the problem with the multilingual forms. Have to say that I did not check yet whether it breaks the single locale title field.

**Comment by NateWr** [comment]
This looks good @ajnyga! I just took it for a spin in IE10/11, Chrome and Firefox and it works pretty well, bar some hiccups that are mostly related to how we handle multi-lingual inputs in general.

In IE 10/11, when tabbing from the Prefix field into the Title field, the inline tools are misplaced, because the fields jump up-and-down with the opening/closing of multilingual inputs:

selection_125

Also, I noticed that tabbing out of the control doesn't close the window. But this seems to be the case for any TinyMCE multingual field with the current `master` branch.

I don't see either of these things as blockers. Do you remember the issue # where we discussed this? I know we had some indexing issues to think about as well.

File: js/controllers/form/MultilingualInputHandler.js (hunk 2 above)
Comment (ajnyga) [reply js/controllers/form/MultilingualInputHandler.js]: Now that I look at it, I have no idea 🤣 

But I removed that and added this: https://github.com/pkp/pkp-lib/pull/3544/files#diff-83df7d7f7c760e14fa91323920849101R235

The problem was that the editor appeared/disappeared as expected, however clicking some button (bold) closed the multilingual popover. So I made the closing js there conditional.

Worked for me, can you check that out?

**Comment by ajnyga** [comment]
@NateWr 
I also noticed that opening the inline editor and then scrolling will move the editor around the page. It has a fixed position. Not much we can do about this I guess?

Check the new version when you have the time.

I can then proceed to editing the form validators (not sure yet if using div as an input field will be a problem) and after that we could implement the changes mentioned on the forum that enable the correct output of italics in titles in different places. The last thing is to take care of the different export plugins.

**Comment by NateWr** [comment]
I just did a quick test and it looks good. :+1: 

> I also noticed that opening the inline editor and then scrolling will move the editor around the page. It has a fixed position. Not much we can do about this I guess?

Yeah I noticed that and I guess it's unfortunate but not a deal-breaker. We could get around it by resetting the position whenever a scroll occurs but it seems like a lot of fragile code for little benefit.

**Comment by ajnyga** [comment]
Great, I will check the form validators next.

**Comment by ajnyga** [comment]
@NateWr, tabbing through the fields should work now. Textareas do not seem to close that way though as you noticed.

**Pull request closed without merging**